src/NuGet.Core/NuGet.Common/Errors/NuGetLogCode.cs

```diff
@@ -640,6 +640,36 @@ public enum NuGetLogCode
         /// </summary>
         NU5035 = 5035,
 
+        /// <summary>
+        /// Error_Maniferst_IconCannotOpenFile
+        /// </summary>
+        NU5036 = 5036,
+
+        /// <summary>
+        /// Error_Maniferst_IconMaxFilesize
+        /// </summary>
+        NU5037 = 5037,
+
+        /// <summary>
+        /// Error_Validation_MultipleIconFiles
+        /// </summary>
+        NU5038 = 5038,
+
+        /// <summary>
+        /// Error_Validation_NoIconFile
+        /// </summary>
+        NU5039 = 5039,
+
+        /// <summary>
+        /// Error_ValidationErrorReadingIcon
+        /// </summary>
+        NU5040 = 5040,
+
+        /// <summary>
+        /// Error_Validation_IconFileEmpty
+        /// </summary>
+        NU5041 = 5041,
```

Comment:
All these error codes feel like an over kill to me. 

It's nice that you are capturing all these scenarios, but i think it's fair to argue that they share an error code.